src/test/java/CityServiceTest.java

```diff
@@ -0,0 +1,175 @@
+        long expectedCountryCode = 1L;
+        City city = cityService.create(expectedCityName, expectedCountryCode);
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenCreateAndCountryCodeIsMissing() {
+        expectedException.expect(InvalidArgumentDhexException.class);
+        String expectedCityName = "Lima";
+        Long expectedCountryCode = null;
+        City city = cityService.create(expectedCityName, expectedCountryCode);
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenCreateAndNameLengthIsInvalid() {
+        expectedException.expect(InvalidArgumentDhexException.class);
+        String expectedCityName = "Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean ma";
+        long expectedCountryCode = 1L;
+        City city = cityService.create(expectedCityName, expectedCountryCode);
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenCreateAndNameExistsForCountryCode() {
+        expectedException.expect(DuplicatedEntityException.class);
+        String expectedCityName = "Lima";
+        long expectedCountryCode = 1L;
+        City city1 = cityService.create(expectedCityName, expectedCountryCode);
+        City city2 = cityService.create(expectedCityName, expectedCountryCode);
+    }
+
+    @Test
+    public void shouldCreateCityWhenCreateAndNameExistsForDifferentCountryCode() {
+        String expectedCityName = "Lima";
+        long expectedCountryCode1 = 1L;
+        long expectedCountryCode2 = 2L;
+        City city1 = cityService.create(expectedCityName, expectedCountryCode1);
+        City city2 = cityService.create(expectedCityName, expectedCountryCode2);
+        assertThat(city1.getId(), is(not(city2.getId())));
+        assertThat(city1.getName(), is(city2.getName()));
+        assertThat(city1.isEnabled(), is(city2.isEnabled()));
+        assertThat(city1.getCountryCode(), is(not(city2.getCountryCode())));
+    }
+
+    // =============
+    // Update tests
+    // =============
+    @Test
+    public void shouldUpdateCityWhenIdExists() {
+        String expectedCityName = "Lima";
+        long expectedCountryCode = 1L;
+        City city = cityService.create(expectedCityName, expectedCountryCode);
+
+        long expectedCityCode = city.getId();
+        boolean expectedNewEnabled = false;
+        city = cityService.update(expectedCityCode, expectedNewEnabled);
+
+        assertThat(city.getId(), is(expectedCityCode));
+        assertThat(city.getName(), is(expectedCityName));
+        assertThat(city.isEnabled(), is(expectedNewEnabled));
+        assertThat(city.getCountryCode(), is(expectedCountryCode));
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenUpdateCityWhenIdDoesNotExist() {
+        expectedException.expect(NotExistingCityException.class);
+        long expectedCityCode = 10L;
+        boolean expectedNewEnabled = false;
+        cityService.update(expectedCityCode, expectedNewEnabled);
+    }
+
+    // =============
```

Comment:
If we realize that our test files are getting large we can split by functionality (CityServiceSearchTest, CityServiceUpdateTest, CityServiceCreateTest...)